firecrown/modeling_tools.py

```diff
@@ -25,13 +25,21 @@ def __init__(
         pt_calculator: Optional[pyccl.nl_pt.EulerianPTCalculator] = None,
         pk_modifiers: Optional[Collection[PowerspectrumModifier]] = None,
         cluster_abundance: Optional[ClusterAbundance] = None,
+        hm_definition: Optional[pyccl.halos.MassDef] = None,
```

Comment:
I think passing the HMCalculator object directly (or a wrapper around it), similar to the `PTCalculator`, would make the API more future proof and also abstract away these CCL details.